### PR TITLE
Added sample I2CMux and OLED driver

### PR DESCRIPTION
**Pull request by SP2FET**
To be verified by the rest of the team :)

### CODE REVIEW

File: Drivers/OLED/lib/ssd1306/ssd1306_conf_template.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+//#define STM32H7
+//#define STM32F7
+
+// Choose a bus
+#define SSD1306_USE_I2C
+//#define SSD1306_USE_SPI
+
+// I2C Configuration
+#define SSD1306_I2C_PORT        hi2c1
+#define SSD1306_I2C_ADDR        (0x3C << 1)
```
Comment (SP2FET) [inline Drivers/OLED/lib/ssd1306/ssd1306_conf_template.h:25]: to be configured

File: Drivers/OLED/I2CMux.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+	//then send data through the channel
+	if (HAL_I2C_Master_Transmit_DMA(&i2cHandle, address, &data, dataSize)
+			!= HAL_OK) {
+		return board::statusERROR;
+	}
+
+	// IMPORTANT! those are only placeholders - proper flow has to be written
+	return board::statusOK;
+}
```
Comment (Lefucjusz) [inline Drivers/OLED/I2CMux.cpp]:
```suggestion
auto I2CMux::send(MuxedI2C targetI2C, std::uint8_t* data,
		std::uint8_t dataSize) -> std::uint8_t {
	// first, select the proper I2C channel in the mux

	//some command to set the proper channel:
	std::uint8_t buffer = 0x01;
	HAL_I2C_Master_Transmit_DMA(&i2cHandle, address, &buffer, sizeof(buffer));

	//then send data through the channel
	if (HAL_I2C_Master_Transmit_DMA(&i2cHandle, address, data, dataSize)
			!= HAL_OK) {
		return board::statusERROR;
	}

	// IMPORTANT! those are only placeholders - proper flow has to be written
	return board::statusOK;
}
```
The second argument should be pointer to uint8_t, so the function can be used to send any type of data of any length.

File: Drivers/OLED/lib/ssd1306/ssd1306_conf_template.h (hunk 1 above)
Comment (SP2FET) [reply Drivers/OLED/lib/ssd1306/ssd1306_conf_template.h:25]: `dupa` test

File: Drivers/OLED/I2CMux.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+// TCA9548A driver
+class I2CMux {
+private:
+I2C_HandleTypeDef i2cHandle;
+std::uint16_t address;
+public:
+	I2CMux(const I2C_HandleTypeDef& _i2cHandle, std::uint16_t _address) : i2cHandle(_i2cHandle),address(_address){};
+	void init();
+	auto send(MuxedI2C targetI2C, std::uint8_t data, std::uint8_t dataSize) -> std::uint8_t;
```
Comment (Lefucjusz) [inline Drivers/OLED/I2CMux.hpp]:
```suggestion
	auto send(MuxedI2C targetI2C, std::uint8_t* data, std::uint8_t dataSize) -> std::uint8_t;
```
Change also in declaration

**Review by Lefucjusz — CHANGES_REQUESTED** [review]

**Review by Lefucjusz — APPROVED** [review]

**Pull request merged**